City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2018: E2E, cleaned up auth_helpers.ts

# [AU-2018](https://helsinkisolutionoffice.atlassian.net/browse/AU-2018)

## What was done

This pull request refactors the`auth_helpers.ts` file. The code has been improved by:
- Splitting functionality into smaller functions.
- Removing duplicate code.
- Adding logging.
- Adding comments.
- Cleaning up the code.

## Other smaller changes
- `form_helpers.ts`: Adjusted `validateFormErrors` to use a locator to prevent flakiness when fetching form errors.
- `playwright.config.ts`: Added a `profiles` test case so that `make test-pw-profiles` works.
- `readme.md`: Fixed a typo.
- `auth.setup.ts`: Removed a line break.
- `e2e.mk`: Modified the `test-pw-profiles` command so that `CREATE_PROFILE` isn't forcefully set.


## How to install

Make sure your instance is up and running on correct branch.
* `git checkout feature/AU-2018-e2e-select-role`
* `make fresh`
* `make drush-cr`

## How to test
- Set `CREATE_PROFILE="TRUE"` in your `.env` file. Run `make test-pw-profiles`. Make sure the test passes.

- Set `CREATE_PROFILE="FALSE"` in your `.env` file. Run `make test-pw-profiles`. Make sure the test passes.

- Delete the `user.json` file inside the `/e2e/.auth/` directory and run `make test-pw-profiles`. Make sure the test passes.

- Run `make test-pw-p PROJECT=forms-48` to make sure form tests still work.

- Review the code changes in `auth_helpers.ts`.


[AU-2018]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2018?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
